src/libraries/System.IO.Compression/src/System/IO/Compression/DeflateZLib/DeflateStream.cs

```diff
@@ -418,6 +422,10 @@ async ValueTask<int> Core(Memory<byte> buffer, CancellationToken cancellationTok
                             int n = await _stream.ReadAsync(new Memory<byte>(_buffer, 0, _buffer.Length), cancellationToken).ConfigureAwait(false);
                             if (n <= 0)
                             {
+                                if (!_inflater.Finished() && _inflater.NonEmptyInput() && _inflater.AvailableOutput > 0)
+                                {
+                                    ThrowGenericInvalidData();
```

Comment:
Does the deflate specification say that we should reject all incomplete decompressions? Is there something similar that we should do on the compression side (Write)?

Reply:
Sorry, I couldn't find this info in the spec (though other implementations like `SharpCompress` and `SharpZipLib` behave like this PR).